Source/PluginProcessor.cpp

```diff
@@ -236,12 +267,27 @@ void PinkTromboneAudioProcessor::processBlock (AudioBuffer<float>& buffer, MidiB
 		}
 	}
 	
-	double constrictionIndex = this->constrictionX * (double) this->tract->getTractIndexCount();
-	double constrictionDiameter = (this->constrictionY/2 + 0.5) * (this->constrictionMax - this->constrictionMin) + this->constrictionMin; //(constrictionY/2 + 0.5) is workaround until AudioParameters are implemented to fix rest diameter range
-	double tongueIndex = tongueX * ((double) (this->tract->tongueIndexUpperBound() - this->tract->tongueIndexLowerBound())) + this->tract->tongueIndexLowerBound();
+	double constrictionIndex;
+	double constrictionDiameter;
+	double tongueIndex;
+	double tongueDiameter;
 	double innerTongueControlRadius = 2.05;
 	double outerTongueControlRadius = 3.5;
-	double tongueDiameter = tongueY * (outerTongueControlRadius - innerTongueControlRadius) + innerTongueControlRadius;
+	
+	if (adsr.isActive()) {

```

Comment:
I don't love that this splits into two code paths here, since basically you're duplicating all this logic. So right now with every call to applyEnvelope you generate an envelope-specific value for any parameter that could be modulated. I might instead just calculate a modulation amount for each parameter. So rather than calculate envelopeConstrictionX, I'd calculate constrictionXModulation. Such a value will be zero when the envelope is idle. Then you don't need a separate code path for when adsr.isActive() is false. Instead you just write something like `constrictionIndex = (*constrictionX + this->constrictionXModulation) * (double) this->tract->getTractIndexCount();`